inference_segmentation.py

```diff
@@ -242,101 +243,140 @@ def classifier(params, img_list, model, device, working_folder):
                    delimiter=',')
 
 
-def main(params: dict):
+def main(params: dict) -> None:
     """
-    Identify the class to which each image belongs.
-    :param params: (dict) Parameters found in the yaml config file.
+    Function to manage details about the inference on segmentation task.
 
+    1. Read the parameters from the config given.
+    2. Read and load the state dict from the previous training or the given one.
+    3. Make the inference on the data specifies in the config.
+
+    -------
+    :param params: (dict) Parameters found in the yaml config file.
     """
-    since = time.time()
+    # since = time.time()
+
+    # PARAMETERS
+    mode = get_key_def('mode', params, expected_type=str)
+    task = get_key_def('task_name', params['task'], expected_type=str)
+    model_name = get_key_def('model_name', params['model'], expected_type=str).lower()
+    num_classes = len(get_key_def('classes_dict', params['dataset']).keys())
+    modalities = read_modalities(get_key_def('modalities', params['dataset'], expected_type=str))
+    BGR_to_RGB = get_key_def('BGR_to_RGB', params['dataset'], expected_type=bool)
+    num_bands = len(modalities)
+    debug = get_key_def('debug', params, default=False, expected_type=bool)
+    # SETTING OUTPUT DIRECTORY
+    try:
+        state_dict = Path(params['inference']['state_dict_path']).resolve(strict=True)
+    except FileNotFoundError:
+        logging.info(
+            f"\nThe state dict path directory '{params['inference']['state_dict_path']}' don't seem to be find," +
+            f"we will try to locate a state dict path in the '{params['general']['save_weights_dir']}' " +
+            f"specify during the training phase"
+        )
+        try:
+            state_dict = Path(params['general']['save_weights_dir']).resolve(strict=True)
+        except FileNotFoundError:
+            raise logging.critical(
+                f"\nThe state dict path directory '{params['general']['save_weights_dir']}'" +
+                f" don't seem to be find either, please specify the path to a state dict"
+            )
```

Comment:
This is user-mistake handling. Worth being discussed I think. From what I understand, if the inputted state_dict is not a file, this script will check the directory where weights should have been saved? 
Then, further down at line 326-338, this state_dict (which should be a directory? it was previously expected to be a path to the checkpoint.pth.tar file directly, no?) is used to find the latest checkpoint.pth.tar file?

If we assume that user can make mistakes of this type (wrong path to an state_dict), it opens the door to a lot more "mistake" handling that we don't currently do. I'd like to discuss this with other team members.